components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/importexport/utils/CommonUtil.java

```diff
@@ -328,4 +331,46 @@ public static String extractArchive(File sourceFile, String destination) throws
             throw new APIImportExportException(errorMessage, e);
         }
     }
+
+    /**
+     * This method will be used to generate Endpoint certificates and meta information related to endpoint certs
+     *
+     * @param filePath String of new file path
+     * @param content  String of content to write into the file
+     * @throws APIManagementException If an error occurs when generating new certs and yaml file
+     */
+    public static void generateFiles(String filePath, String content) throws APIManagementException {
+
+        File file = new File(filePath);
+        try (FileOutputStream fos = new FileOutputStream(file)) {
+
+            if (!file.exists()) {
+                file.createNewFile();
+            }
+            byte[] bytesArray = content.getBytes();
+
+            fos.write(bytesArray);
+            fos.flush();
+
+        } catch (IOException e) {
+            String errorMessage = "Error while generating meta information of client certificates from path.";
+            throw new APIManagementException(errorMessage, e);
```

Comment:
Shall we add a finally clause here? 

```
        finally {
            try {
                if (fos != null) {
                    fos.close();
                }
            } catch (IOException e) {
                log.error("Error occurred closing file output stream", e);
            }
        }
```

Reply:
The current code uses the Java 7 `try with resources` pattern which will cause `fos` to be closed automatically when it goes out of scope. Therefore it does not require to be explicitly closed in a finally block.